### DISCUSSION TITLE
[Resolved] Multiple command rules by pressing a button in succession [Resolved]

### DISCUSSION THREAD

**Discussion by simone1970** [arendst/Tasmota#15939] [Q&A] [answered]
Good morning everyone.

I can't create a rule that allows you to execute multiple commands by pressing a single button in succession ...

Example

I press button 1 and the command publish "xxxxxxxx" message "A" starts
I press button 1 again, the command publish "xxxxxxxx" message "B" starts
I press button 1 again, the command publish "xxxxxxxx" message "C" starts

Can anyone explain to me how I could do it ??

I apologize for any English translation errors ....

Thank you...

**Comment by sfromis** [comment]
See https://tasmota.github.io/docs/Buttons-and-Switches/#multi-press-functions

**Reply by simone1970** [reply]
Thanks for the reply....

I had already seen the link that you kindly indicated to me, but if I'm not mistaken it talks about multi-pressing of the button, that is I press once and a command is executed, I press twice in a row and another command is executed etc ...

I would like a rule where you press the button and a command is executed, you press the button again (but once) and a new command is executed ...


I hope I was clear with the translation ...

**Reply by pkkrusty** [reply]
If it's only three things, you could turn on/off the rulesets in succession. Rule1 On switch#state do backlog publish something; rule2 1; rule1 0 endon
Rule2 On switch#state do backlog publish something; rule3 1; rule2 0 endon 

Etc...

**Comment by barbudor** [comment, marked as answer]
Using variable as state machine 
```
Rule1
  on system#boot do var1 0 endon
  On button#state do add1 1 endon
  On var1#state==1 do publish some/topic somevalue endon
  On var1#state==2 do publish some/topic somevalue endon
  On var1#state==3 do backlog var1 0; publish some/topic somevalue endon
```

**Reply by simone1970** [reply]
You are fantastic.  Works!!!!!!!!!!!!!!!!!!!!!
Thank you.

**Reply by chuckhathundra** [reply]
great, that's what I was looking for. is it possible that one click will be long, thanks

**Reply by barbudor** [reply]
https://tasmota.github.io/docs/Buttons-and-Switches/#multi-press-functions

**Comment by chuckhathundra** [comment]
i need to set the value   On var1#state11==1 do publish some/topic somevalue endon

Rule1
  on system#boot do var1 0 endon
  On button#state do add1 1 endon
  On var1#state11==1 do publish some/topic somevalue endon
  On var1#state==2 do publish some/topic somevalue endon
  On var1#state==3 do backlog var1 0; publish some/topic somevalue endon

**Reply by barbudor** [reply]
> On var1#state11==1 do publish some/topic somevalue endon

Totally wrong syntax on this line while you 2 next lines are correct


You are very unclear about what you want to achieve.
You first talk about long press which would be triggered by `button#state==3`
Now you talk about values of a variable

**Reply by sfromis** [reply]
Do you mean `var11#state==1` or `var1#state==11` ? That would be valid trigger syntax.

**Comment by chuckhathundra** [comment]
I don't know how to insert one command with a long hold into this list of commands.
Rule1
on system#boot do var1 0 endon
On button#state do add1 1 endon
On var1#(HOLD BUTTON PRESS )==1 do publish some/topic somevalue endon
On var1#state==2 do publish some/topic somevalue endon
On var1#state==3 do backlog var1 0; publish some/topic somevalue endon

**Reply by sfromis** [reply]
`Button#state==3` is when a long press of 4 seconds was detected. See
https://tasmota.github.io/docs/Buttons-and-Switches/#multi-press-functions

**Comment by chuckhathundra** [comment]
Thank you all for your attention and patience with me. I will try to explain my wish. There is an apparatus, it has one button, the button does such a job.
One click - turns on,
second press - turns on the timer,
the third press is to add one timer step, 
long press is to turn on the interval, 
the second long press is to turn on another interval mode. The first part of the code works well. How to insert code on long press of a button?
